tools/runner/logger.go

```diff
@@ -0,0 +1,98 @@
+package runner
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	grpcv1 "github.com/grpc/test-infra/api/v1"
+	"github.com/grpc/test-infra/status"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PodLogger provides functionality to save pod logs to files.
+type PodLogger struct {
+}
+
+// NewPodLogger creates a new PodLogger object.
+func NewPodLogger() *PodLogger {
+	return &PodLogger{}
+}
+
+// savePodLogs saves pod logs to files with same name as pod.
+func (pl *PodLogger) savePodLogs(ctx context.Context, loadTest *grpcv1.LoadTest, podLogDir string) error {
+	// Get pods for this test
+	pods, err := pl.getTestPods(ctx, loadTest)
+	if err != nil {
+		return err
+	}
+
+	// Attempt to create directory. Will not error if directory already exists
+	err = os.MkdirAll(podLogDir, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("Failed to create pod log output directory %s: %v", podLogDir, err)
+	}
+
+	// Write logs to files
+	errorOccured := false
```

Comment:
"Occurred".

I think we should consider fetching the log to memory and then writing it if it is not empty. Logs in our case are small (or empty), and this would separate two classes of errors, one from getting the logs from the pods, and the other from writing the logs to the files.